mor/mor/feel/feelmor/toolboxmor_impl.hpp

```diff
@@ -271,27 +409,76 @@ ToolboxMor<SpaceType, Options>::assembleData()
     for( int i = 0; i < M_F; ++i )
         this->M_Fqm[0][0][i] = qf[i];
 
+    auto u = this->Xh->element();
+
     // output
-    // form1(_test=this->Xh, _vector=this->M_Fqm[1][0][0])
-    //     += integrate( _range=markedelements( this->Xh->mesh(), "IC2" ), _expr= id( v ) );
+    auto outputs = M_modelProperties->outputs();
+    int output = 1;
+    for( auto const& outp : outputs )
+    {
+        auto out = outp.second;
+        if( out.type() == "average" )
+        {
+            auto dim = out.dim();
+            auto fAvg = form1(_test=this->Xh);
+            if( dim == nDim )
+            {
+                auto range = markedelements(this->Xh->mesh(), out.markers());
+                double area = integrate( _range=range, _expr=cst(1.0) ).evaluate()(0,0) ;
+                fAvg += integrate( _range=range, _expr=id(u)/cst(area) );
+            }
+            else if( dim == nDim-1 )
+            {
+                auto range = markedfaces(this->Xh->mesh(), out.markers());
+                double area = integrate( _range=range, _expr=cst(1.0) ).evaluate()(0,0) ;
+                fAvg += integrate( _range=range, _expr=id(u)/cst(area) );
+            }
+            this->M_Fqm[output++][0][0] = fAvg.vectorPtr();
+        }
+        else if( out.type() == "flux")
```

Comment:
this is the normal_derivative, not the flux
the flux would have the conduction coefficient -k*grad(u)*N